json-rpc/src/runtime.rs

```diff
@@ -138,6 +178,14 @@ async fn rpc_endpoint_without_metrics(
         .get_latest_ledger_info()
         .map_err(|_| reject::custom(DatabaseError))?;
 
+    let mut rng = OsRng;
+    let trace_id = rng.next_u64();
+    let request_log = RpcRequestLog {
+        trace_id,
+        request: data.clone(),
+    };
+    info!(request_log);
```

Comment:
Minor, you can put this into the macro, and then if the log level doesn't log this, it won't create the struct and do the cloning of data.